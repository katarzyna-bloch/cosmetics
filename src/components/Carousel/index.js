import React from 'react'
import styled from 'styled-components'
import Slider from 'react-slick'
import 'slick-carousel/slick/slick.css'
import 'slick-carousel/slick/slick-theme.css'
import PropTypes from 'prop-types'

const Container = styled.div`
  width: 100%;
  margin-bottom: ${props => (props.withBottomMargin ? '32px' : 0)};

  .slick-slider {
    width: 100%;
    max-width: calc(100vw - 50px);
  }

  .slick-arrow.slick-prev {
    left: 18px;
    z-index: 1;
  }
  .slick-arrow.slick-next {
    right: 18px;
    z-index: 1;
  }

  ${({ sideMargins }) => sideMargins && `
    .slick-list {
      margin-left: -${sideMargins}px;
      margin-right: -${sideMargins}px;
    }
    .slick-slide > div {
      padding-left: ${sideMargins}px;
      padding-right: ${sideMargins}px;
    }
  `};
`

const Slide = styled.div``

export const Carousel = ({ items, settings = {} }) => {
  const defaultSettings = {
    dots: false,
    infinite: true,
    speed: 200,
    draggable: false,
    gap: 0,
  }

  const dots = items.length < 10 ? settings.dots || defaultSettings.dots : false
  const gap = settings.gap || defaultSettings.gap

  return (
    <Container withBottomMargin={dots} sideMargins={gap / 2}>
      <Slider {...defaultSettings} {...settings} dots={dots}>
        {items.map((item, index) => (
          <Slide key={index}>
            {item}
          </Slide>
        ))}
      </Slider>
    </Container>
  )
}

Carousel.propTypes = {
  items: PropTypes.arrayOf(PropTypes.element.isRequired).isRequired,
  settings: PropTypes.object.isRequired,
}

export default Carousel
