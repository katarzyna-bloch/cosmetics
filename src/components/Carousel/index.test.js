import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Carousel from '.'

jest.mock('react-slick', () => {
  const Slider = () => 'Slider'
  return Slider
})

it('should render correctly', () => {
  const Banner = ({ name }) => <div>Banner: {name}</div>
  const items = [
    <Banner name="Banner 1" />,
    <Banner name="Banner 2" />,
    <Banner name="Banner 3" />,
  ]
  
  const props = {
    settings: {},
    items,
  }
  const wrapped = shallow(<Carousel {...props}/>)
  expect(toJson(wrapped)).toMatchSnapshot()
})
