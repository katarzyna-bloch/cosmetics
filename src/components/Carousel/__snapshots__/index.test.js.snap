// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<styled.div
  sideMargins={0}
  withBottomMargin={false}
>
  <Slider
    dots={false}
    draggable={false}
    gap={0}
    infinite={true}
    speed={200}
  >
    <styled.div
      key="0"
    >
      <Banner
        name="Banner 1"
      />
    </styled.div>
    <styled.div
      key="1"
    >
      <Banner
        name="Banner 2"
      />
    </styled.div>
    <styled.div
      key="2"
    >
      <Banner
        name="Banner 3"
      />
    </styled.div>
  </Slider>
</styled.div>
`;
